server/src/main/java/com/linecorp/line/auth/fido/fido2/server/service/ResponseServiceImpl.java

```diff
@@ -300,13 +300,14 @@ protected void verifySignature(ServerAuthPublicKeyCredential serverPublicKeyCred
         byte[] signatureBytes = Base64.getUrlDecoder().decode(serverPublicKeyCredential.getResponse().getSignature());
         boolean result = SignatureHelper.verifySignature(userKey.getPublicKey(), toBeSignedMessage, signatureBytes, userKey.getAlgorithm());
         if (!result) {
-            throw new FIDO2ServerRuntimeException(InternalErrorCode.ASSERTION_SIGNATURE_VERIFICATION_FAIL);
```

Comment:
I'm not sure whether we should return aaguid for errors encountered after decoding authenticator data.

Based on my understanding, there are also other kind of failures like
- user handle mismatch
- RP ID hash mismatch
- flag not set etc.

I would prefer also setting aaguid for these cases.

Reply:
Thank you for your review. 
Most of the code has been modified based on your comments. Please check again.